timemachine/fe/rbfe.py

```diff
@@ -1,102 +1,109 @@
-        intg_impl,
-        bound_impls,
-        baro_impl,
-    )
+    x0 = ligand_conf
+    box0 = np.eye(3, dtype=np.float64) * 10  # make a large 10x10x10nm box
 
-    # burn-in
-    ctxt.multiple_steps_U(
-        n_steps=protocol.n_eq_steps,
-        store_u_interval=0,
-        store_x_interval=0,
-    )
+    return x0, box0, hmr_masses, potentials, baro
 
-    assert np.all(np.isfinite(ctxt.get_x_t())), "Equilibration resulted in a nan"
 
-    # a crude, and probably not great, guess on the decorrelation time
-    n_steps = protocol.n_frames * protocol.steps_per_frame
-    _, all_coords, all_boxes = ctxt.multiple_steps_U(
-        n_steps=n_steps,
-        store_u_interval=0,
-        store_x_interval=protocol.steps_per_frame,
-    )
+def setup_in_env(st, host, host_config, ligand_conf, lamb, temperature, run_seed):
+
+    host_system, host_masses, host_conf = host
+
+    system = st.combine_with_host(host_system, lamb=lamb)
+    combined_masses = np.concatenate([host_masses, st.combine_masses()])
+
+    potentials = system.get_U_fns()
+    hmr_masses = model_utils.apply_hmr(combined_masses, system.bond.get_idxs())
+
+    group_idxs = get_group_indices(get_bond_list(system.bond))
+    baro = MonteCarloBarostat(len(hmr_masses), DEFAULT_PRESSURE, temperature, group_idxs, 15, run_seed + 1)
+
+    x0 = np.concatenate([host_conf, ligand_conf])
+    box0 = host_config.box
+
+    return x0, box0, hmr_masses, potentials, baro
+
 
-    assert all_coords.shape[0] == protocol.n_frames
-    assert all_boxes.shape[0] == protocol.n_frames
+def assert_all_states_have_same_masses(initial_states: List[InitialState]):
+    # hmr masses should be identical throughout the lambda schedule
+    # bond idxs should be the same at the two end-states, note that a possible corner
+    # case with bond breaking may seem to be problematic:
 
-    assert np.all(np.isfinite(all_coords[-1])), "Production resulted in a nan"
+    # 0 1 2    0 1 2
+    # C-O-C -> C.H-C
 
-    return all_coords, all_boxes
+    # but this isn't an issue, since hydrogens will only ever be terminal atoms
+    # and core hydrogens that are mapped to heavy atoms will take the mass of the
+    # heavy atom (thereby not triggering the mass repartitioning to begin with).
 
+    # but it's reasonable to be skeptical, so we also assert consistency through the lambda
+    # schedule as an extra sanity check.
+    masses = np.array([s.integrator.masses for s in initial_states])
+    deviation_among_windows = masses.std(0)
+    np.testing.assert_array_almost_equal(deviation_among_windows, 0, err_msg="masses assumed constant w.r.t. lambda")
 
-# setup the initial state so we can (hopefully) bitwise recover the identical simulation
-# to help us debug errors.
-def setup_initial_states(st, host_config, temperature, lambda_schedule, seed, min_cutoff=0.7):
+
+def setup_initial_states(
+    st,
+    host_config,
+    temperature,
+    lambda_schedule,
+    seed,
+    min_cutoff=np.inf,
```

Comment:
why did the default change here?

Reply:
To suppress a spurious error in debugging -- will revert to 0.7

Reply:
Reverted in https://github.com/proteneer/timemachine/pull/936/commits/c4319ba50409589a413859b6568184f06991bdb1